### DISCUSSION TITLE
Powerstream

### DISCUSSION THREAD

**Discussion by homecoq** [tolwi/hassio-ecoflow-cloud#238] [Q&A]
Good morning,

No monitoring of the powerstream temperature and inverter relay status is KO?

is a repair planned?

**Comment by Nid01** [comment]
@homecoq 
I'm receiving inverter relay data, but like noted in your issue https://github.com/tolwi/hassio-ecoflow-cloud/issues/239 the temperature data is currently not being provided by the API.
![image](https://github.com/tolwi/hassio-ecoflow-cloud/assets/30300755/7518b018-960d-4828-abfa-34327069f7d8)
I suggest closing the question here so the gathered information can be focused in the issue.

**Comment by homecoq** [comment]
Hello,

The temperature data worked at one time and still works but in a very random way

![Temperatures](https://github.com/tolwi/hassio-ecoflow-cloud/assets/65351083/0d30f996-f7b0-44ac-82de-84acf0cb9d40)

**Reply by Nid01** [reply]
> Hello,
> 
> The temperature data worked at one time and still works but in a very random way
> 
> ![Temperatures](https://private-user-images.githubusercontent.com/65351083/336121292-0d30f996-f7b0-44ac-82de-84acf0cb9d40.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1MzE5OTksIm5iZiI6MTcxNzUzMTY5OSwicGF0aCI6Ii82NTM1MTA4My8zMzYxMjEyOTItMGQzMGY5OTYtZjdiMC00NGFjLTgyZGUtODRhY2YwY2I5ZDQwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA0VDIwMDgxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkNzExZjJiZGQ5M2JmMmI2Yzk5ODJlYzhlN2FmMWVkOGQ2ZjM0YzUxMjM1NjVlOWU2OThkNWFkMjkzNDM0YmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J-WpqsilwxmDO1SbMBozUvQLqv0cina1lQUnuynCIAs)

@homecoq
Sorry, but you are showing the temperature data of the solar panel connected to PowerStream.
From just two values I can't understand what you mean by very random.
For May I have this readings which don't look that unusual to me:
![image](https://github.com/tolwi/hassio-ecoflow-cloud/assets/30300755/8949d095-73f9-4ed6-b132-1584246cf1a7)

**Comment by granny-82** [comment]
Hello, first of all thanks for implementing this feature <3 

I have installed my inverter yesterday and setup the home assistant integration today. The inverter temperature was collected without any issues. I have then updated the inverter from 1.0.0 to 1.0.1 and finally to 1.0.1.213  

After those firmware updates the temperatur sensor is no longer displayed. Maybe this information helps you.

**Reply by Nid01** [reply]
> Hello, first of all thanks for implementing this feature <3
> 
> I have installed my inverter yesterday and setup the home assistant integration today. The inverter temperature was collected without any issues. I have then updated the inverter from 1.0.0 to 1.0.1 and finally to 1.0.1.213
> 
> After those firmware updates the temperature sensor is no longer displayed. Maybe this information helps you.

Thank you for your information.
I now have some e-mail correspondence with EcoFlow Customer Service Team and their latest response from 18.07.2024 was:
> From the feedback of our senior group, we are afraid this very field is not open to public at the moment ,we will bring it to our users in future.
I now have the chance to ask, why they seemingly dropped the data point in the newer firmware version.

**Reply by giovanne123** [reply]
> > Hello, first of all thanks for implementing this feature <3
> > I have installed my inverter yesterday and setup the home assistant integration today. The inverter temperature was collected without any issues. I have then updated the inverter from 1.0.0 to 1.0.1 and finally to 1.0.1.213
> > After those firmware updates the temperature sensor is no longer displayed. Maybe this information helps you.
> 
> Thank you for your information. I now have some e-mail correspondence with EcoFlow Customer Service Team and their latest response from 18.07.2024 was:
> 
> > From the feedback of our senior group, we are afraid this very field is not open to public at the moment ,we will bring it to our users in future.
> > I now have the chance to ask, why they seemingly dropped the data point in the newer firmware version.

Ask them, I am interested in temperature too, like it was already working before (for solar1/2 ). 
Someone can get the feeling that they intentionally no longer publish the temperature (for solar1/2) because they know it get's to hot and therefore they now purchased their Smart Fan for Powerstream to prevent that 😉

**Comment by granny-82** [comment]
I have also noted a very strange behaviour that the power output seems to be limited to strictly 400 watts. This might also be related to the high temperatures. I have no covered the inverter so it is no longer directly in the sunlight. This flat tops are no longer visible. I want to collect more data thought. Right now this is just an assumption with too little sunny days. Just wanted to update you. 

Above: Inverter in the shade but overall a sometimes cloudy day 
Below: Inverter in direct sunlight
![image](https://github.com/user-attachments/assets/e8ae9153-6bcb-4ebf-b84d-bde209e91ef6)

Any update/reply yet from ecoflow? thanks :)

**Reply by Nid01** [reply]
My latest reply from EcoFlow was:

> Sorry to keep you waiting for so long!
> The temperature API is temporarily suspended and will be reopened later, but the exact time for reopening has not been determined yet.
> We apologize for any inconvenience this may cause during the process.
> Have a nice day!

Since the integration is currently implementing the public API and also the public API since July  provides the sensor "espTempsensor" I  notified the support, that this sensor is what I was looking for and closed my case.
![image](https://github.com/user-attachments/assets/1e2846d5-2d33-4342-b4fc-8b6fd84a2318)

**Reply by yongliqi** [reply]
Is this espTempsensor still providing data?

**Reply by Nid01** [reply]
> Is this espTempsensor still providing data?

Yes, I still can read temperature data from it.

**Reply by yongliqi** [reply]
Thank you. From long time ago, I only got -1. Any clue? Temp from panels is ok though.

**Reply by Nid01** [reply]
> Thank you. From long time ago, I only got -1. Any clue? Temp from panels is ok though.

Sorry. No, I have no idea what could be the issue when you are using the public API and still receive -1 for the sensor iot.espTempsensor.
